mir-ci/mir_ci/test_drag_and_drop.py

```diff
@@ -40,27 +53,43 @@ class TestDragAndDrop:
         ],
     )
     async def test_source_and_dest_match(self, modern_server, app) -> None:
-        modern_server = DisplayServer(modern_server, add_extensions=VirtualPointer.required_extensions)
-        program = modern_server.program(apps.App(app))
+        extensions = VirtualPointer.required_extensions + ScreencopyTracker.required_extensions
+        server_instance = DisplayServer(modern_server, add_extensions=extensions)
+        program = server_instance.program(apps.App(app))
+
+        robot_settings = dedent(
+            f"""\
+            Library    {MIR_CI_PATH}/robot_libraries/WaylandHid.py
+            Library    {MIR_CI_PATH}/robot_libraries/Screencopy.py
+            Resource   {MIR_CI_PATH}/robot_resources/screencopy.resource
+        """
+        ).strip("\n")
 
-        robot_test_case = f"""Source and Destination Match
-            Sleep     {STARTUP_TIME}
-            Move Pointer To Absolute    40    40
-            Sleep     {A_SHORT_TIME}
-            Press LEFT Button
-            Sleep     {A_SHORT_TIME}
-            Move Pointer To Absolute    120    70
-            Sleep     {A_SHORT_TIME}
-            Move Pointer To Absolute    200    100
-            Sleep     {A_SHORT_TIME}
-            Release LEFT Button
+        robot_test_case = dedent(
+            f"""\
+            Source and Destination Match
+                Sleep     {STARTUP_TIME}
+                ${{regions}} =    Test Match    {MIR_CI_PATH}/robot_templates/drag_and_drop_src.png
+                ${{center}} =    Get Center    ${{regions}}[0]
+                Move Pointer To Absolute    ${{center}}[x]    ${{center}}[y]
+                Sleep     {A_SHORT_TIME}
+                Press LEFT Button
+                ${{off_center}} =    Add Displacement    ${{center}}    20    20
+                Move Pointer To Absolute    ${{off_center}}[x]    ${{off_center}}[y]
+                Sleep     {A_SHORT_TIME}
+                ${{regions}} =    Test Match    {MIR_CI_PATH}/robot_templates/drag_and_drop_dst.png
+                ${{center}} =    Get Center    ${{regions}}[0]
+                Move Pointer To Absolute    ${{center}}[x]    ${{center}}[y]
+                Sleep     {A_SHORT_TIME}
```

Comment:
How do we get rid of those sleeps? Add a hover effect and wait on that? After clicking, the background goes blue, so we could wait for that?

Reply:
We can simply remove them (and make the tests way faster). I believe those sleeps are intended to make it easier for us to see each action. They are part of the original code.

Reply:
Oh the original code had them b/c that was the only way for us to get things in the right order and for the app to react to input. Note this may run on slower hardware (e.g. Pi4) and we can't be sure the app registers input as fast as we send it. I'd say drop them, and we'll see what our automation makes of it.

When we get videos out if it is when we'll see things in (slowed down) action.

Reply:
Some tests fail when they are removed. I've created an issue on this one.